lib/resourcemerge/core.go

```diff
@@ -164,6 +164,10 @@ func ensureProbePtr(modified *bool, existing **corev1.Probe, required *corev1.Pr
 
 func ensureProbe(modified *bool, existing *corev1.Probe, required corev1.Probe) {
 	setInt32(modified, &existing.InitialDelaySeconds, required.InitialDelaySeconds)
+	setInt32(modified, &existing.TimeoutSeconds, required.TimeoutSeconds)
+	setInt32(modified, &existing.PeriodSeconds, required.PeriodSeconds)
+	setInt32(modified, &existing.SuccessThreshold, required.SuccessThreshold)
+	setInt32(modified, &existing.FailureThreshold, required.FailureThreshold)
```

Comment:
I double-checked [the `Probe` type][1], and with these four additions we are now completely covering the type.  The current master logic is unchanged since probe handling landed in d9f6718de07 (#7), and that commit doesn't provide motivation for ignoring the properties you're adding here.

[1]: https://github.com/openshift/cluster-version-operator/blob/fc25a6f43fcb4a0025585de51cb55190037f168f/vendor/k8s.io/api/core/v1/types.go#L2010-L2036